src/sentry/similarity/index.py

```diff
@@ -27,6 +28,38 @@ def __init__(self, cluster, namespace, signature_builder, bands, interval, reten
         self.interval = interval
         self.retention = retention
 
+    def __build_signatures(self, items):
```

Comment:
This is the meat and potatoes of the Python changes, it collects an iterable of feature sets into a structure of this form:

```
Map[
  bytes,  # index id
  List[   # bands
    Map[
      bytes,  # band identifier
      int,  # number of times seen
    ]
  ]
]
```

and then transforms it into the form that is used by the Lua script, which is:

```
<number of indexes to parse> [<index id> [<number of buckets in band> [<band identifier> <number of times seen> …] …] …]
```